src/lsqecc/gates/compress_rotation_approximations.py

```diff
@@ -8,6 +8,13 @@ def partition_gate_sequence(gate_approximation: str):
     for index, gate in enumerate(gate_approximation):
         if gate in {"S", "T"}:
             partition += gate
+            if index == len(gate_approximation) - 1:
+                if not in_x_basis:
+                    partitioned_sequence.append(partition)
+                else:
+                    partitioned_sequence.append(partition[0])
+                    partitioned_sequence.append(partition[1:])
+
```

Comment:
You didn't like [the concise implementation based on `.split("H")`](https://replit.com/@gwwatkin/partitiongatesequence)? I thought it would have been cleaner, there is quite a bit of logic here  :sweat_smile: .

Note that you would need to add X handling, but I think you could add it pretty concisely and get the 'HXH' case which might be nice to have since the gate sequences keep changing

Reply:
Ok I see some of the subtlety now, you would also have to modify the split solution based to get the behavoiur you described in your comment. I think you can decide best which to go with :slightly_smiling_face:

Reply:
The concise implementation was nice :) I read the comments from bottom-up, last time! so by the time I got to it, I already fixed some bugs and was just too lazy to change it!

Reply:
I tried modifying the split behavior this morning, but it would require gate-by-gate processing like the code already exists, and becomes as complex as the code that exists to my knowledge and the edge cases are not handled properly or rather I am not able to come up with a cleaner solution! 
If we want to modify cases like `HSSSS` or while encountering an `X`. The main idea behind my code, I keep adding to a partition/subsequence as long as I am in the same basis. Once I encounter a change of basis, either an X or a second H, I pop out the subsequence/partition to the list of partitions.